gengokit/httptransport/httptransport.go

```diff
@@ -381,6 +399,10 @@ func createDecodeConvertFunc(f Field) (string, bool) {
 var {{.LocalName}} *{{.GoType}}
 {{.LocalName}} = &{{.GoType}}{}
 err = json.Unmarshal([]byte({{.LocalName}}Str), {{.LocalName}})`
+		singlePBTypeUnmarshalTmpl := `
+var {{.LocalName}} *{{.GoType}}
+{{.LocalName}} = &{{.GoType}}{}
+err = jsonpb.UnmarshalString({{.LocalName}}Str, {{.LocalName}})`
```

Comment:
Instead of creating thses tailored decode functions for each individual field, how about we we first merge all the variables from path, query, and body into a single JSON string, then call `jsonpb.Unmarshal(s, req)` to decode into the `request` struct in one go? Seems like it would simplify the decode logic significantly?